src/components/App.jsx

```diff
@@ -1,80 +1,182 @@
-import React, { useState, useEffect } from 'react';
-import $ from 'jquery';
+import React, { useState, useEffect, useReducer } from 'react';
+import PhotoSet from './PhotoSet.jsx';
+
+//takes about a second to transition and about every 5-8 seconds is when it does it.
+
 
 function App() {
+
+  const initialState = {previous: [], current: [], next: [], mainPhoto: 0} 
+
+  function reducer (state, action) { //reducer and initial state are required for creating a reduce hook
+    let photoNum; //state always refers to the initialState object
+    switch (action.type) { //switch is basically a fancy if statement
+      case 'next': //basically this is if(action.type === 'next')
+        let array = makePhotoArray((state.mainPhoto + 2) % photos.length)
+        if (state.mainPhoto === photos.length - 1) {
+          photoNum = 0
+        } else {
+          photoNum = state.mainPhoto + 1
+        }
+        return {
+          previous: state.current,
+          current: state.next,
+          next: makePhotoArray((state.mainPhoto + 2) % photos.length),
+          mainPhoto: state.mainPhoto + 1
+        };
+      case 'previous':
+        if (state.mainPhoto === 0) {
+          photoNum = photos.length - 1
+        } else {
+          photoNum = state.mainPhoto - 1
+        }
+        return {
+          previous: makePhotoArray((photos.length + (state.mainPhoto - 2)) % photos.length),
+          current: state.previous,
+          next: state.current,
+          mainPhoto: state.mainPhoto - 1
+        };
+      default:
+        if (state.previous.length > 0 || photos.length === 0) {
+          return state;
+        } else {
+          return {
+            previous: makePhotoArray(photos.length - 1),
+            current: makePhotoArray(0),
+            next: makePhotoArray(1),
+            mainPhoto: 0
+          }
+        }
+    }
+  }
+
   const [restaurantId, updateId] =  useState(null); 
   const [photos, updatePhotos] = useState([]);
-  const [visiblePhotos, updateVisbles] = useState([]);
-  const [currentPhoto, updateCurrent] = useState(0);
+  const [photosOfInterest, dispatch] = useReducer(reducer, initialState);
+  const [stopwatch, updateWatch] = useState(false);
 
   useEffect(() => { //Loads a random restaurant ID upon loading the page, will be replaced when all modules are comined
     let newId = Math.floor(Math.random() * (100));
     updateId(newId);
   }, []) //Runs every page load
   
   useEffect(() => { //Loads the photos from said restaurant Id
-    $.ajax({
-      method: "GET",
-      url: `/api/photo-gallery-list/${restaurantId}`,
-      contentType: "application/json",
-      success: (restData) => {
+    fetch(`/api/photo-gallery-list/${restaurantId}`)
```

Comment:
I think in the future we should communicate on which library we want to use for ajax just so that when we compile everything together we're not using a mix of jquery, axios, fetch, etc.

Reply:
Sounds good to me.  Sorry for the mismatching.